sql/_13_issues/_23_1h/cases/cbrd_24843_2.sql

```diff
@@ -0,0 +1,647 @@
+    c.*
+from
+    t_child as c,
+    t_parent as p
+where
+    c.p_a = p.a and c.p_b = p.b
+    and (c.p_a = 5 or c.p_b = 5);
+show trace;
+
+
+/* 6. '(p.a = 6 or p.b = 6)' */
+/* ansi‑style */
+select /*+ recompile */
+    c.*
+from
+    t_parent as p
+    inner join t_child as c on c.p_a = p.a and c.p_b = p.b
+where
+    (p.a = 6 or p.b = 6);
+show trace;
+
+select /*+ recompile */
+    c.*
+from
+    t_child as c,
+    t_parent as p
+where
+    c.p_a = p.a and c.p_b = p.b
+    and (p.a = 6 or p.b = 6);
+show trace;
+
+
+/* 7. 'p.a = cast (7 as numeric (1, 0))' */
+/* ansi‑style */
+select /*+ recompile */
+    c.*
+from
+    t_parent as p
+    inner join t_child as c on c.p_a = p.a and c.p_b = p.b
+where
+    p.a = cast (7 as numeric (1, 0));
+show trace;
+
+select /*+ recompile */
+    c.*
+from
+    t_child as c,
+    t_parent as p
+where
+    c.p_a = p.a and c.p_b = p.b
+    and p.a = cast (7 as numeric (1, 0));
+show trace;
+
+
+/* 8. 'p.a >= 8' */
+/* ansi‑style */
+select /*+ recompile */
+    c.*
+from
+    t_parent as p
+    inner join t_child as c on c.p_a = p.a and c.p_b = p.b
+where
+    p.a >= 8;
+show trace;
+
+select /*+ recompile */
+    c.*
+from
+    t_child as c,
+    t_parent as p
+where
+    c.p_a = p.a and c.p_b = p.b
+    and p.a >= 8;
+show trace;
+
+
+/* 9. 'p.a in (1, 3, 5, 7, 9)' */
+/* ansi‑style */
+select /*+ recompile */
+    c.*
+from
+    t_parent as p
+    inner join t_child as c on c.p_a = p.a and c.p_b = p.b
+where
+    p.a in (1, 3, 5, 7, 9);
+show trace;
+
+select /*+ recompile */
+    c.*
+from
+    t_child as c,
+    t_parent as p
+where
+    c.p_a = p.a and c.p_b = p.b
+    and p.a in (1, 3, 5, 7, 9);
+show trace;
+
+
+/* 10. 'p.a between 2 and 4' */
+/* ansi‑style */
+select /*+ recompile */
+    c.*
+from
+    t_parent as p
+    inner join t_child as c on c.p_a = p.a and c.p_b = p.b
+where
+    p.a between 2 and 4;
+show trace;
+
+select /*+ recompile */
+    c.*
+from
+    t_child as c,
+    t_parent as p
+where
+    c.p_a = p.a and c.p_b = p.b
+    and p.a between 2 and 4;
+show trace;
+
+
+/* 11. 'p.a = (select a from t_parent order by a desc limit 1)' */
+/* ansi‑style */
+select /*+ recompile */
+    c.*
+from
+    t_parent as p
+    inner join t_child as c on c.p_a = p.a and c.p_b = p.b
+where
+    p.a = (select a from t_parent order by a desc limit 1);
+show trace;
+
+select /*+ recompile */
+    c.*
+from
+    t_child as c,
+    t_parent as p
+where
+    c.p_a = p.a and c.p_b = p.b
+    and p.a = (select a from t_parent order by a desc limit 1);
+show trace;
+
+
+/* 12. 'c.p_a <> 1' -> pt_lambda_node() -> 'p.a <> 1' */
+/*     'c.p_a != 1' -> pt_lambda_node() -> 'p.a != 1' */
+/* ansi‑style */
+select /*+ recompile */
+    c.*
+from
+    t_parent as p
+    inner join t_child as c on c.p_a = p.a and c.p_b = p.b
+where
+    c.p_a <> 1;
+show trace;
+
+select /*+ recompile */
+    c.*
+from
+    t_child as c,
+    t_parent as p
+where
+    c.p_a = p.a and c.p_b = p.b
+    and c.p_a <> 1;
```

Comment:
```suggestion
    and p.a <> 1;
```
The purpose of this test group is to test what pt_lambda_node() does and what doesn't.
For this purpose it should be tested with p.a rather than c.p_a.

Reply:
review is reflected.